lucet-runtime/lucet-runtime-internals/src/alloc/mod.rs

```diff
@@ -154,25 +154,25 @@ impl Alloc {
         // the above makes sure this expression does not underflow
         let guard_remaining = self.heap_inaccessible_size - expand_pagealigned as usize;
 
-        let heap_spec = module.heap_spec();
-        // The compiler specifies how much guard (memory which traps on access) must be beyond the
-        // end of the accessible memory. We cannot perform an expansion that would make this region
-        // smaller than the compiler expected it to be.
-        if guard_remaining < heap_spec.guard_size as usize {
-            bail_limits_exceeded!("expansion would leave guard memory too small");
-        }
+        if let Some(heap_spec) = module.heap_spec() {
```

Comment:
The expand heap opcode is guaranteed not to exist, thanks to wasm module validation, but a hostcall could attempt to expand the heap, which should return an error rather than silently do nothing. I'll fix this.